### PR TITLE
feat: pre-download the ingested sst

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Pre-download the ingested SST files from remote object store to fill the local file cache to accelerate following reads.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/mito2/src/cache/write_cache.rs [hunk 1]
```diff
@@ -166,6 +166,76 @@ impl WriteCache {
...
     }
 
+    /// Downloads a file in `remote_path` from the remote object store to the local cache
+    /// (specified by `index_key`).
+    pub(crate) async fn download(
+        &self,
+        index_key: IndexKey,
+        remote_path: &str,
+        remote_store: &ObjectStore,
+    ) -> Result<()> {
```
Comment (evenyag) [inline src/mito2/src/cache/write_cache.rs:176]: Can we add a timer to record the download time?

File: src/mito2/src/cache/write_cache.rs [hunk 2]
```diff
@@ -166,6 +166,76 @@ impl WriteCache {
...
+        writer.close().await.context(error::DownloadSnafu {
+            region_id,
+            file_id,
+            file_type,
+        })?;
+
+        DOWNLOAD_BYTES_TOTAL.inc_by(bytes_written);
+
+        debug!(
+            "Successfully download file '{}' to local '{}', region: {}, cost: {:?}",
```
Comment (evenyag) [inline src/mito2/src/cache/write_cache.rs:225]: Also log file size

File: src/mito2/src/metrics.rs [hunk 3]
```diff
@@ -189,6 +189,12 @@ lazy_static! {
         &[TYPE_LABEL]
     )
     .unwrap();
+    /// Download bytes counter.
+    pub static ref DOWNLOAD_BYTES_TOTAL: IntCounter = register_int_counter!(
+        "mito_download_bytes_total",
```
Comment (evenyag) [inline src/mito2/src/metrics.rs:194]: Maybe rename to `mito_write_cache_download_bytes_total`

File: src/mito2/src/metrics.rs [hunk 4]
```diff
@@ -189,12 +189,17 @@ lazy_static! {
...
-    )
-    .unwrap();
+    /// Download bytes counter in the write cache.
+    pub static ref WRITE_CACHE_DOWNLOAD_BYTES_TOTAL: IntCounter = register_int_counter!(
+        "mito_write_cache_download_bytes_total",
+        "mito write cache download bytes total",
+    ).unwrap();
+    /// Timer of the downloading task in the write cache.
+    pub static ref WRITE_CACHE_DOWNLOAD_ELAPSED_TOTAL: HistogramVec = register_histogram_vec!(
+        "mito_write_cache_download_elapsed_total",
```
Comment (evenyag) [inline src/mito2/src/metrics.rs:199]:
```suggestion
    pub static ref WRITE_CACHE_DOWNLOAD_ELAPSED: HistogramVec = register_histogram_vec!(
        "mito_write_cache_download_elapsed",
```

**Review by evenyag — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**